src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/DevirtualizationManager.cs

```diff
@@ -15,6 +15,11 @@ public DevirtualizationManager(CompilationModuleGroup compilationModuleGroup)
             _compilationModuleGroup = compilationModuleGroup;
         }
 
+        public override bool IsEffectivelySealed(TypeDesc type)
+        {
+            return !(type is RuntimeDeterminedType) && base.IsEffectivelySealed(type);
```

Comment:
Do you happen to know why we are not hitting this case in the full AOT CoreRT compilation (and thus why this case is not handled by the base class)?

Reply:
Not really but I can find out by temporarily commenting out the change. I believe this is a problem I hit while R2R compiling one of the framework assemblies so an obvious difference is that we're compiling the CoreCLR framework assemblies here.

Reply:
OK, I'm seeing this crash the build of one framework assembly, System.Threading.Tasks.Parallel. Here the method is called from getClassAttrib on the type "TLocal_System.__Canon" while compiling the method

<code>
[System.Threading.Tasks.Parallel]System.Threading.Tasks.Parallel+<>c__DisplayClass20_0`1<TLocal_System.__Canon>.<ForWorker64>b__1(RangeWorker&,int32,bool&)
</code>

I tried to repro this by compiling the library in the multifile mode but I haven't hit this particular check, maybe due to some differences in dependency rooting between the R2R and RyuJit mode.

Reply:
> is called from getClassAttrib on the type "TLocal_System.__Canon"

That kind of looks like we passed a runtime determined type to ryujit over JitInterface. That should not happen. Ryujit should never see runtime determined types because it doesn't know what to do with them. They only exist in the dependency analysis system (outside project x that is)

Reply:
I have reverted this change in 3rd commit. I'm going to investigate separately why the one particular framework assembly fails compiling. I have also added more fixes for PE rewriting - we shouldn't copy over the import, export and IAT directories because they confuse the Windows PE loader and end up throwing an exception while loading a R2R DLL.